src/Message/PurchaseRequest.php

```diff
@@ -29,88 +33,1392 @@ class PurchaseRequest extends AbstractRequest
+     * @return self
+     * @throws InvalidParameterException if $value is invalid.
+     */
+    public function set3DsRequestAuthenticationMethod($value)
+    {
+        if (in_array(
+            $value,
+            [
+                null,
+                self::ACCOUNT_AUTHENTICATION_METHOD_NONE,
+                self::ACCOUNT_AUTHENTICATION_METHOD_OWN_CREDENTIALS,
+                self::ACCOUNT_AUTHENTICATION_METHOD_FEDERATED_ID,
+                self::ACCOUNT_AUTHENTICATION_METHOD_ISSUER_CREDENTIALS,
+                self::ACCOUNT_AUTHENTICATION_METHOD_THIRD_PARTY_AUTHENTICATION,
+                self::ACCOUNT_AUTHENTICATION_METHOD_FIDO,
+            ]
+        )) {
+            return $this->setParameter('3DsRequestAuthenticationMethod', $value);
+        }
+        throw new InvalidParameterException("Invalid 3DsRequestAuthenticationMethod parameter");
+    }
+
+    /**
+     * Get the 3DsRequestAuthenticationTime field
+     *
+     * Corresponds to the Ds_Merchant_Emv3Ds.threeDSRequestorAuthenticationInfo.threeDSReqAuthTimestamp field in Redsys
+     * documentation.
+     *
+     * @return DateTime|int
+     */
+    public function get3DsRequestAuthenticationTime()
+    {
+        return $this->getParameter('3DsRequestAuthenticationTime');
+    }
+
+    /**
+     * Set the 3DsRequestAuthenticationTime field
+     *
+     * Corresponds to the Ds_Merchant_Emv3Ds.threeDSRequestorAuthenticationInfo.threeDSReqAuthTimestamp field in the
+     * Redsys documentation.
+     *
+     * @param DateTime|int $value
+     * @return self
+     */
+    public function set3DsRequestAuthenticationTime($value)
+    {
+        return $this->setParameter('3DsRequestAuthenticationTime', $value);
+    }
+
+    /**
+     * Get the customerAccountCreatedIndicator field
+     *
+     * Corresponds to the Ds_Merchant_Emv3Ds.acctInfo.chAccAgeInd field in Redsys documentation.
+     *
+     * @return string One of the CUSTOMER_ACCOUNT_CREATED_* constants
+     */
+    public function getCustomerAccountCreatedIndicator()
+    {
+        return $this->getParameter('customerAccountCreatedIndicator');
+    }
+
+    /**
+     * Set the customerAccountCreatedIndicator field
+     *
+     * Corresponds to the Ds_Merchant_Emv3Ds.acctInfo.chAccAgeInd field in the Redsys documentation.
+     *
+     * @param string $value One of the CUSTOMER_ACCOUNT_CREATED_* constants
+     * @return self
+     * @throws InvalidParameterException if $value is invalid.
+     */
+    public function setCustomerAccountCreatedIndicator($value)
+    {
+        if (in_array(
+            $value,
+            [
+                null,
+                self::CUSTOMER_ACCOUNT_CREATED_NONE,
+                self::CUSTOMER_ACCOUNT_CREATED_THIS_TRANSACTION,
+                self::CUSTOMER_ACCOUNT_CREATED_LAST_30_DAYS,
+                self::CUSTOMER_ACCOUNT_CREATED_LAST_60_DAYS,
+                self::CUSTOMER_ACCOUNT_CREATED_MORE_THAN_60_DAYS,
+                ]
```

Comment:
Outdent - should have been picked up by the linter? Diito 696 and 698 and a few other functions as well.